### DISCUSSION TITLE
Regularization

### DISCUSSION THREAD

**Discussion by spatel81** [Nek5000/nekRS#599] [General]
Im solving a LES problem that includes a temperature field. In my par file, I set the following

`regularization = hpfrt + nModes=1 + scalingCoeff=10  `...under the `[GENERAL]` section. Does this apply the high pass [filter](https://nek5000.github.io/NekDoc/problem_setup/filter.html) to all fields, including temperature? 

If I were to add `regularization = avm` to the `[TEMPERATURE]` section, does that mean an additional viscous term (based on the paper by Lu 2019?) is treated, on top of the hpfrt?

**Comment by stgeke** [comment]
The regularization setting in `[GENERAL]` holds for all fields, but you can overwrite it in a field section. In your example you would use hpfRT to velocity and avm for temperature.

**Reply by spatel81** [reply]
I see. so it would not be both for my case - just avm for temp?

**Reply by stgeke** [reply]
Yes, just avm (not both) for temp.

**Reply by spatel81** [reply]
what is the difference between applying avm using the par file vs. applying avm via the .udf with uservp (e.g. [here](https://github.com/Nek5000/nekRS/blob/next/examples/turbPipe/turbPipe.udf))?

**Reply by stgeke** [reply]
If applied manually (not in par), you gain greater flexibility. For example, you could multiply it with an additional coefficient or base it on another quantity.
